src/base_model.jl

```diff
@@ -75,12 +74,36 @@ function build_base_model!(ECModel::AbstractEC, optimizer; use_notations=false)
         max(P_P_us_overestimate[u, t], P_N_us_overestimate[u, t])  # Max between the maximum values calculated previously
     )
 
+    # Overestimation of the thermal power exchanged by each "user" when uploading to the "Thermal Grid" by each user
+    @expression(model_user, P_P_us_overestimate_th[u in user_set, t in time_set],
+        max(0,
+            # sum(Float64[field_component(users_data[u], g, "max_capacity") 
+            #     for g in asset_names(users_data[u], THER)]) # Maximum capacity of thermal generator
+            # + sum(Float64[field_component(users_data[u], r, "max_capacity")*profile_component(users_data[u], r, "ren_pu")[t] 
+            #       for r = asset_names(users_data[u], REN)]) # Maximum dispatch of renewable assets as solar panels
+            + sum(Float64[field_component(users_data[u], g, "max_capacity")*field_component(users_data[u], g, "max_technical")
+                for g = asset_names(users_data[u], THER)]) #Maximum dispatch of the fuel-fired generators
+            - sum(Float64[profile_component(users_data[u], l, "t_load")[t] for l in asset_names(users_data[u], LOAD)]) # Minimum demand
+        ) * TOL_BOUNDS
+    )
+
+    # Overestimation of the thermal power exchanged by each "user" when downloading from the "Thermal Grid" by each user
+    @expression(model_user, P_N_us_overestimate_th[u in user_set, t in time_set],
+        max(0,
+            sum(Float64[profile_component(users_data[u], l, "t_load")[t] for l in asset_names(users_data[u], LOAD)])
+                # Maximum thermal demand
+            + sum(Float64[field_component(users_data[u], g, "max_capacity") 
+                for g in asset_names(users_data[u], THER)])  # Maximum capacity of thermal generator
+            + sum(Float64[field_component(users_data[u], s, "max_capacity") 
+                for s in asset_names(users_data[u], STOR)])  # Maximum capacity of thermal storage
+        ) * TOL_BOUNDS
+    )
 
     ## Variable definition
     
     # Energy stored in the battery
     @variable(model_user, 
-        0 <= E_batt_us[u=user_set, b=asset_names(users_data[u], BATT), t=time_set] 
+        0 <= E_batt_us[u=user_set, b=asset_names(users_data[u], STOR), t=time_set] 

```

Comment:
Please restore to BATT as this block models the electric batteries.
A new object for STOR is required with similar formulation but for TES. It is better to use TES instead of STOR